src/RewardsStreamerMP.sol

```diff
@@ -487,16 +487,17 @@ contract RewardsStreamerMP is
         return accruedMP;
     }
 
-    function _updateAccountMP(address accountAddress) internal {
+    function _updateAccountMP(address accountAddress, bool forceMPUpdate) internal {
         Account storage account = accounts[accountAddress];
         uint256 accruedMP = _getAccountPendingdMP(account);
-
-        account.mpAccrued += accruedMP;
-        account.lastMPUpdateTime = block.timestamp;
+        if (accruedMP > 0 || forceMPUpdate) {
+            account.mpAccrued += accruedMP;
+            account.lastMPUpdateTime = block.timestamp;
+        }
```

Comment:
Different options here... I went with `forceMPUpdate` flag, instead of explicitly updating `account.lastMPUpdateTime` in places where necessary.